### PR TITLE
Update ODE Java Version to Java 21

### PR DESCRIPTION
**Pull request by mwodahl**
This PR Updates the Java version used by the ODE to the latest long-term support version of Java (21)

These changes have been tested by running the ODE application in the devcontainer and by examining the output of mvn clean package

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ X ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

- [ X ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X ] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [ X ] I have added tests to cover my changes.
- [ X ] All new and existing tests passed.

### CODE REVIEW

**Review by drewjj — CHANGES_REQUESTED** [review]
Everything functions with a docker-compose. The tests pass if I manually run them. Sonar seems to fail building the jpo-ode-common project though. Complaining about incompatibilities with Java 21. This may be a sonar configuration issue but I don't think it would be good to entirely break sonar unless we know what is going on.

File: jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/upload/FileUploadIntegrationTests.java [hunk 1]
```diff
@@ -1,84 +0,0 @@
...
-import org.springframework.test.context.junit4.SpringRunner;
-import org.springframework.util.LinkedMultiValueMap;
-import org.springframework.util.MultiValueMap;
-import org.springframework.web.multipart.MultipartFile;
-
-import us.dot.its.jpo.ode.storage.StorageService;
-
-@RunWith(SpringRunner.class)
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
-public class FileUploadIntegrationTests {
```
Comment (dmccoystephenson) [inline jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/upload/FileUploadIntegrationTests.java:44]: Why was this file removed?

**Review by dmccoystephenson — APPROVED** [review]
This looks good!

Verified the following:
- Zookeeper/kafka still works in dev container
- Unit tests pass
- Program starts up without issue

I left a few clarifying questions also.

File: jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/upload/FileUploadIntegrationTests.java (hunk 1 above)
Comment (drewjj) [reply jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/upload/FileUploadIntegrationTests.java:44]: I removed this due to it being a test that has been failing for a while now. It isn't something that can easily be reworked due to it being an integration test. This repository has poor code coverage anyways, so I would prioritize functioning tests.

**Comment by iyourshaw** [comment]
The sonar log has this error thrown by the maven compiler plugin:

> Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project jpo-ode-common: Fatal error compiling: error: release version 21 not supported

That's happening at the compilation step, before Sonar is trying to scan or anything.

I think the java version it runs with is set in `.github/workflows/ci.yaml`, isn't it? That still has Java 11 configured.  Have you tried changing that to 21?

https://github.com/CDOT-CV/jpo-ode/blob/update/java-lts-version/.github/workflows/ci.yml

current contents:

```yaml
name: ci

on: [pull_request, push]

jobs:
  docker:
    runs-on: ubuntu-latest
    steps:
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2
      - name: Build
        uses: docker/build-push-action@v3

  sonar:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          submodules: recursive
      - name: Set up JDK
        uses: actions/setup-java@v3
        with:
          java-version: "11"
          distribution: "temurin"
      - name: Run Sonar
        env:
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        run: |
          ls -la && pwd
          mvn -e -X clean org.jacoco:jacoco-maven-plugin:prepare-agent package sonar:sonar -Dsonar.projectKey=usdot.jpo.ode:jpo-ode -Dsonar.projectName=jpo-ode -Dsonar.organization=usdot-jpo-ode -Dsonar.host.url=https://sonarcloud.io -Dsonar.branch.name=$GITHUB_REF_NAME
```

**Review by drewjj — APPROVED** [review]
This is good now. Same old sonar failure we have seen for a long time now.
